org.sonarlint.eclipse.pydev/src/org/sonarlint/eclipse/pydev/internal/PythonProjectConfiguratorExtension.java

```diff
@@ -72,4 +78,24 @@ public Optional<IDocumentPartitioner> documentPartitioner(String ruleLanguage) {
     }
     return Optional.empty();
   }
+
+  /**
+   *  This is independent from PyDev itself but linked here as it is for Python itself! Currently there is no direct
+   *  way to read about virtual environments from
+   */
+  @Override
+  public Set<IPath> getExclusions(IProject project) {
+    var exclusions = new HashSet<IPath>();
+
+    // Python virtual environments can be named optionally, therefore we only can "guess" the default names based on
+    // different libraries / tools that create virtual environments. We will never catch all probably, but for users
+    // naming their virtual environments differently that is their own "fault".
+    // INFO: These is not the Java standard library PATH class, but the one from Eclipse that is called the same!
```

Comment:
This would be a later option yes, this is the babystep. For your idea I think there is something on Productboard to narrow down the focus based on the user preferences - currently not the case and not possible on any flavor but wished for by quite some users.

Reply:
Added the ticket here: https://sonarsource.atlassian.net/browse/SLE-932